micropip/_compat_in_pyodide.py

```diff
@@ -21,6 +22,24 @@
         raise
     # Otherwise, this is pytest test collection so let it go.
 
+if IN_BROWSER:
+
+    async def _pyfetch(url: str, **kwargs):
+        if "signal" in kwargs:
+            return await pyfetch(url, **kwargs)
+
+        controller = AbortController.new()
+        kwargs["signal"] = controller.signal
+
+        try:
+            return await pyfetch(url, **kwargs)
+        except CancelledError:
+            controller.abort()
+            raise
```

Comment:
Would be great to implement this into pyfetch directly in a followup.

Reply:
So should I remove changes in this file and open another PR on `pyfetch` in pyodide/pyodide?

Do you think this feature should provide a parameter for user to opt-out this behavior? Maybe adding a kw-only parameter like `abort_on_cancel=True`?

Reply:
No, let's add it here and also upstream. 

> Maybe adding a kw-only parameter like `abort_on_cancel=True`?

Nah it feels like we're switching from a less correct behavior to a more correct one with this. Let's not overcomplicate the API by forcing people to opt into what they expect.